typed-store/src/rocks/mod.rs

```diff
@@ -204,15 +204,12 @@ impl DBBatch {
         self.batch.delete_range_cf(&db.cf(), from_buf, to_buf);
         Ok(self)
     }
-}
 
-impl DBBatch {
     /// inserts a range of (key, value) pairs given as an iterator
-    #[allow(clippy::map_collect_result_unit)] // we don't want a mutable argument
-    pub fn insert_batch<K: Serialize, V: Serialize, T: Iterator<Item = (K, V)>>(
+    pub fn insert_batch<J: Borrow<K>, K: Serialize, U: Borrow<V>, V: Serialize>(
```

Comment:
If I am now mistaken we can also change the type of pretty much all other functions to only take any type that can be a ref to K instead of K or &K right? Is there a point to that?